File: core/src/main/java/xtdb/operator/IRelationSelector.java
package xtdb.operator;

import org.apache.arrow.memory.BufferAllocator;
import xtdb.vector.RelationReader;

public interface IRelationSelector {
    /**
     * @param params a single-row indirect relation containing the params for this invocation - maybe a view over a bigger param relation.
     */
    int[] select(BufferAllocator allocator, RelationReader readRelation, RelationReader params);
}

File: core/src/main/kotlin/xtdb/operator/IRelationSelector.kt
package xtdb.operator

import clojure.lang.IPersistentMap
import org.apache.arrow.memory.BufferAllocator
import xtdb.vector.RelationReader

interface IRelationSelector {
    /**
     * @param params a single-row indirect relation containing the params for this invocation - maybe a view over a bigger param relation.
     */
    fun select(allocator: BufferAllocator, readRelation: RelationReader, schema: IPersistentMap, params: RelationReader): IntArray
}
